docs/reference/search/term-enum.asciidoc

```diff
@@ -0,0 +1,126 @@
+[[search-term-enum]]
+=== Term enum API
+
+The term enum API can be used to discover terms in the index that match
+a pattern. This can be useful for features like auto-complete or regexp authoring tools:
+
+[source,console]
+--------------------------------------------------
+POST stackoverflow/_terms/list
+{
+    "field" : "tags",
+    "pattern" : "kiba"
+}
+--------------------------------------------------
+// TEST[setup:stackoverflow]
```

Comment:
An Interesting API, is it supposed to be externally facing as well? Or only be used internally for suggestions etc?

Reply:
> is it supposed to be externally facing as well? 

I'd like it to be public facing - it seems perverse to hide the terms that prefix, wildcard and regexp queries use under the covers. Why shouldn't we be transparent about what's in the index? 
Would stop questions [like this](https://discuss.elastic.co/t/help-elasticsearch-regexp-query/253445) which I've seen many times.